app/src/main/kotlin/com/strv/archdemo/ui/counter/CounterViewModel.kt

```diff
@@ -1,19 +1,40 @@
 package com.strv.archdemo.ui.counter
 
 import android.os.Parcelable
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import com.strv.archdemo.extension.toBundle
+import com.strv.archdemo.ktools.Preferences
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.parcelize.Parcelize
+import javax.inject.Inject
 
 @Parcelize
 data class CounterConfig(val text: String) : Parcelable
 
-class CounterViewModel(
-    val savedStateHandle: SavedStateHandle
+@HiltViewModel
+class CounterViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    val preferences: Preferences,
 ) : ViewModel() {
     private val args = CounterFragmentArgs.fromBundle(savedStateHandle.toBundle())
 
     // We should always hold all data (state) inside ViewModel
     val title = args.config.text
+    val clickCount = MutableLiveData(preferences.getCounterValueFromPreferences())
```

Comment:
I'm not sure what are you planning to show in this demo, so it's difficult to judge, but are `LiveData` the best option?

They quite nicely remove the need to unsubscribe from `ViewModel` once `onDestroyView` is called. However this might be an important concept to explain, so if you want to avoid using `StateFlow` maybe simple `ObservableField` would do?

Reply:
We decided to use `MutableStateFlow` as we are be showing Coroutines usage anyway.